numpy/core/src/common/simd/avx2/conversion.h

```diff
@@ -29,4 +29,17 @@
 #define npyv_cvt_b32_f32(BL) _mm256_castps_si256(BL)
 #define npyv_cvt_b64_f64(BL) _mm256_castpd_si256(BL)
 
+// expand
+NPY_FINLINE void npyv_expand_u8_u16(npyv_u8 data, npyv_u16 *low, npyv_u16 *high) {
+    const __m256i z = _mm256_setzero_si256();
+    *low = _mm256_unpacklo_epi8(data, z);
```

Comment:
you can't use [_mm256_unpacklo_epi8()](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm256_unpacklo_epi8&expand=1609,6093) without shuffle the odd lanes since its interleave each 128-bit lane, you should use `_mm256_cvtepu8_epi16` instead.

please add testing unit for any new intriniscs so we can discover issues like this through CI since currently all builders using the default baseline.
I think we need to add CI builds with diffenrent sets of baseline features, I will release a pr for it.
The last think I think the signature of this intrinisc should be :
`NPY_FINLINE npyv_u16x2 npyv_expand_u8_u16(npyv_u8 a)`

`npyv_u16x2` is more frinedly with the compilers and the our testing unit.

Reply:
I will add testing unit, and the new signature looks fine to me.